cli/client.go

```diff
@@ -710,17 +710,13 @@ func (c *viamClient) robot(orgStr, locStr, robotStr string) (*apppb.Robot, error
 		return nil, err
 	}
 
-	robots, err := c.listRobots(orgStr, locStr)
+	resp, err := c.client.GetRobot(c.c.Context, &apppb.GetRobotRequest{
+		Id: robotStr,
+	})
 	if err != nil {
 		return nil, err
 	}
-
-	for _, robot := range robots {
-		if robot.Id == robotStr || robot.Name == robotStr {
-			return robot, nil
-		}
-	}
```

Comment:
As @zaporter-work pointed out and unless I'm missing something, I think we don't want to get rid of this functionality. It could just look for `robot.Name` matches, since you queried ID with `GetRobot` above.